src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -271,6 +271,7 @@ public static void Move(string sourceDirName, string destDirName)
             bool sameDirectoryDifferentCase =
                 directoriesAreCaseVariants &&
                 destDirNameFromFullPath.Equals(sourceDirNameFromFullPath, fileSystemSensitivity);
+            bool sourcePathIsFile = FileSystem.FileExists(fullsourceDirName);
```

Comment:
Why call it this early? Consider moving this new call right before the `if` where it is first used (line 289).